libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -384,60 +383,6 @@ func (n *bridgeNetwork) setDefaultForwardRule(
 	return nil
 }
 
-type iptRule struct {
-	ipv   iptables.IPVersion
-	table iptables.Table
-	chain string
-	args  []string
-}
-
-// Exists returns true if the rule exists in the kernel.
-func (r iptRule) Exists() bool {
-	return iptables.GetIptable(r.ipv).Exists(r.table, r.chain, r.args...)
```

Comment:
I was slightly trying to grasp the reason behind this change; because previously this was non-exported and internal to the bridge driver. Is the intent to make this type more widely used, and used as an abstraction for iptables + nftables?

In that case, I'm wondering if it's currently the right abstraction, or at least, I'm wondering if all fields it defines are "portable";

- Should there be an interface defined instead (providing `Exists()`, `Append()`, `Insert()`, `Delete()`)
- The fields to be unexported, and instead having a constructor (`iptables.NewRule(...)`)
- Alternatively, the `iptables` package to have `Delete(rule)`, `Insert(rule)` functions (which could potentially become an abstraction?)

Do you have more context to this change?

Reply:
The end goal is to decouple the bridge driver from iptables.

I have a wip branch where I'm moving iptables rules into a new internal package. There will be transitional commits where both the `bridge` package and the new `fwipt` pacakge will need this struct.

`iptRule` is currently the best iptables abstraction we have. It'd be great to converge on that and drop some of the alternative functions we've (eg. `ProgramRule`). 

> and used as an abstraction for iptables + nftables?

No, we won't use the same abstraction for both, they're too different. So, no need to worry about 'portability'.

Reply:
> iptRule is currently the best iptables abstraction we have

Yeah, the `args` feels like a leaky abstraction; the combination of args (doing the actual work) with some defined fields, is a bit odd 😂